packages/caravan-health/src/feescore.ts

```diff
@@ -0,0 +1,145 @@
+import { BlockchainClient } from "@caravan/clients";
+import { utxoSetLengthScore } from "./privacy";
+import { Transaction, AddressUtxos } from "./types";
+
+/*
+Utility function that helps to obtain the fee rate of the transaction
+
+Expected Range : [0, 0.75]
+-> Very Poor : [0, 0.15]
+-> Poor : (0.15, 0.3]
+-> Moderate : (0.3, 0.45]
+-> Good : (0.45, 0.6]
+-> Very Good : (0.6, 0.75]
+*/
+function getFeeRateForTransaction(transaction: Transaction): number {
+  let fees: number = transaction.fee;
+  let weight: number = transaction.weight;
+  return fees / weight;
+}
+
+/* 
+Utility function that helps to obtain the percentile of the fees paid by user in tx block
+
+Expected Range : [0, 0.75]
+-> 0% tile : 1
+-> 10% tile : 0.9
+-> 25% tile : 0.75
+-> 50% tile : 0.5
+-> 75% tile : 0.25
+-> 90% tile : 0.1
+-> 100% tile : 0.05
+*/
+async function getFeeRatePercentileScore(
+  timestamp: number,
+  feeRate: number,
+  client: BlockchainClient,
```

Comment:
I'm on the fence on this but my feeling is that decoupling from the client completely might be preferable by simply taking a feeRatePercentile as an argument and returning the score. It slightly complicates the use of this function, but you can add that as a docstring describing the function. If someone wants to find fee rate percentile on their own they can.